api/common/v1alpha1/ref.go

```diff
@@ -46,6 +54,9 @@ type TargetRef struct {
 	Tags map[string]string `json:"tags,omitempty"`
 	// Mesh is reserved for future use to identify cross mesh resources.
 	Mesh string `json:"mesh,omitempty"`
+	// ProxyTypes specifies the data plane types that are subject to the policy. When not specified,
+	// all data plane types are targeted by the policy.
```

Comment:
Would be cool if we could have this plus `omitempty`

```suggestion
	// all data plane types are targeted by the policy.
	// +kubebuilder:validation:MinItems=1
```

Reply:
Changed, looks like works 👍 

```➜  kumactl git:(add-proxy-types-targetref) ✗ echo "
type: MeshAccessLog
mesh: default
name: mal
spec:
  targetRef:
    kind: Mesh
    proxyTypes: []
  to:
    - targetRef:
        kind: Mesh
      default:
        backends:
          - type: File
            file:
              path: /dev/stdout" | ./kumactl apply -f -
Error: YAML contains invalid resource: spec.targetRef.proxyTypes: in body should have at least 1 items
➜  kumactl git:(add-proxy-types-targetref) ✗ echo "
type: MeshAccessLog
mesh: default
name: mal
spec:
  targetRef:
    kind: Mesh
  to:
    - targetRef:
        kind: Mesh
      default:
        backends:
          - type: File
            file:
              path: /dev/stdout" | ./kumactl apply -f -
➜  kumactl git:(add-proxy-types-targetref) ✗```